crates/ruff_python_formatter/src/string/implicit.rs

```diff
@@ -303,25 +331,53 @@ impl Format<PyFormatContext<'_>> for FormatImplicitConcatenatedStringFlat<'_> {
                 StringLikePart::FString(f_string) => {
                     for element in &f_string.elements {
                         match element {
-                            FStringElement::Literal(literal) => {
+                            FTStringElement::Literal(literal) => {
                                 FormatLiteralContent {
                                     range: literal.range(),
                                     flags: self.flags,
-                                    is_fstring: true,
+                                    is_ftstring: true,
                                     trim_end: false,
                                     trim_start: false,
                                 }
                                 .fmt(f)?;
                             }
                             // Formatting the expression here and in the expanded version is safe **only**
                             // because we assert that the f-string never contains any comments.
-                            FStringElement::Expression(expression) => {
-                                let context = FStringContext::new(
+                            FTStringElement::Expression(expression) => {
+                                let context = FTStringContext::new(
+                                    self.flags,
+                                    FTStringLayout::from_f_string(f_string, f.context().source()),
+                                );
+
+                                FormatFTStringInterpolatedElement::new(expression, context)
+                                    .fmt(f)?;
+                            }
+                        }
+                    }
+                }
+                StringLikePart::TString(f_string) => {
```

Comment:
The code duplication here seems unnecessary (and it uses `f-string` instead of `t-string`). Maybe another place where an `FTString` enum could help.